### ISSUE TITLE
SqlServiceAccount not working for SSRS 2016

### ISSUE THREAD

**Issue by rliggett** [PowerShell/SqlServerDsc#1259]
**Problem:**

SqlServiceAccount does not work for SSRS 2016, it produced an internal error.

**Description**

We have successfully changed the service account for Database Services (DatabaseEngine) using SqlServiceAccount however i get an internal error when i try to set the service account for SSRS 2016.

**Steps to produce error**

SSRS 2016 EE on Windows Server 2016

$cred = Get-Credential -UserName "testdomain.local\testaccount" -Message "Password please"

Invoke-DSCResource -Module SqlServerDsc -Name SqlServiceAccount -Method Set -Property @{
serverName='TestServer''
InstanceName='TESTSSRS'
ServiceType='ReportingServices'
ServiceAccount=$cred
}

**Error Produced**

An internal error occurred. 
    + CategoryInfo          : NotSpecified: (root/Microsoft/...gurationManager:String) [], CimException
    + FullyQualifiedErrorId : HRESULT 0x8007054f
    + PSComputerName        : localhost

**Exploration**

I have debugged the code and see that it errors out at SetServiceAccount. ( I also tried SetServiceAccount in standard powershell which also doesn't work for SSRS, but again works  fine for database engine)

It looks like SetServiceAccount is not valid for reporting services and we should use SetWindowsServiceIdentity, a method of MSReportServer_ConfigurationSetting.

Has anyone successfully used SqlServiceAccount for SSRS 2016.

**Comment by johlju** [comment]
@rliggett I don't think this resource has been tested at all with reporting services. So I think you are on the right track that we should use something else to change the service account for that service so that Reporting Services is aware of the change.

Labeled this a bug and help wanted so that anyone in the community can run with this issue.

**Comment by shurick81** [comment]
Here is some info about encryption key manipulation when changing service account name: https://docs.microsoft.com/en-us/sql/reporting-services/install-windows/ssrs-encryption-keys-back-up-and-restore-encryption-keys?view=sql-server-2017

**Comment by shurick81** [comment]
Here is a POC, works in my dev environment: https://github.com/shurick81/Dynamics365Configuration/blob/fd6895ed0591faac80fe36bb7bfd46e840b22475/infrastructure/stacks/rsserviceaccountupdate.ps1

**Comment by johlju** [comment]
There are some information in closed duplicate issue https://github.com/dsccommunity/SqlServerDsc/issues/1463 that could be relevant.

**Comment by johlju** [comment]
It is clearly state in the documentation that one should not change the account directly in the `Services.msc`.
Mentioned in [Best practices for changing the service account for the report server in SQL Server Reporting Services](https://learn.microsoft.com/en-us/troubleshoot/sql/reporting-services/best-practices-change-service-account) and [Changing the Service Account](https://learn.microsoft.com/en-us/sql/reporting-services/install-windows/configure-the-report-server-service-account-ssrs-configuration-manager#changing-the-service-account).

The resource SqlServiceAccount would probably not even work as I think it should call the method [`SetWindowsServiceIdentity()`](https://learn.microsoft.com/en-us/sql/reporting-services/wmi-provider-library-reference/configurationsetting-method-setwindowsserviceidentity?view=sql-server-ver16) if the service is Reporting Services.
